Lib/test/datetimetester.py

```diff
@@ -1224,6 +1236,15 @@ def test_roundtrip(self):
             dt2 = self.theclass(dt.year, dt.month, dt.day)
             self.assertEqual(dt, dt2)
 
+    def test_repr_subclass(self):
+        name = SubclassDate.__name__
+        td = SubclassDate(1, 2, 3)
+        self.assertEqual(repr(td), "%s(1, 2, 3)" % name)
+        td = SubclassDate(2014, 1, 1)
+        self.assertEqual(repr(td), "%s(2014, 1, 1)" % name)
+        td = SubclassDate(2010, 10, day=10)
+        self.assertEqual(repr(td), "%s(2010, 10, 10)" % name)
```

Comment:
Let's hard-code the expectation (here and throughout the tests). The extra indirection, while reducing some repetition, makes it more difficult to read what's being asserted, but also could mask a surprised change if `SubclassDate.__name__` somehow changes.

```suggestion
        td = SubclassDate(1, 2, 3)
        self.assertEqual(repr(td), "SubclassDate(1, 2, 3)")
        td = SubclassDate(2014, 1, 1)
        self.assertEqual(repr(td), "SubclassDate(2014, 1, 1)")
        td = SubclassDate(2010, 10, day=10)
        self.assertEqual(repr(td), "SubclassDate(2010, 10, 10)")
```